src/SmartComponents/ProviderPage/providerForm.js

```diff
@@ -1,141 +1,138 @@
+    name: stepName,
+    stepKey: stepName,
+    nextStep
+});
+
+const temporaryHardcodedSourceSchemas = {
+    openshift: {
+        title: 'Configure OpenShift',
+        /* fields: [{
+            component: componentTypes.TEXT_FIELD,
+            name: 'role',
+            type: 'hidden'
+        }, */
+        fields: [{
+            component: componentTypes.TEXT_FIELD,
+            name: 'url',
+            label: 'URL'
+        }, {
+            component: componentTypes.CHECKBOX,
+            name: 'verify_ssl',
+            label: 'Verify SSL'
+        }, {
+            component: componentTypes.TEXT_FIELD,
+            name: 'certificate_authority',
+            label: 'Certificate Authority',
+            condition: {
+                when: 'verify_ssl',
+                is: true
+            }
+        }, {
+            component: componentTypes.TEXTAREA_FIELD,
+            name: 'token',
+            label: 'Token'
+        }]
     },
-    uiSchema: {
-        password: { 'ui:widget': 'password' }
+    amazon: {
+        title: 'Configure AWS',
+        fields: [{
+            component: componentTypes.TEXT_FIELD,
+            name: 'user_name',
+            label: 'Access Key'
+        }, {
+            component: componentTypes.TEXT_FIELD,
+            name: 'password',
+            label: 'Secret Key'
+        }]
     }
 };
 
-import { componentTypes, validatorTypes } from '@data-driven-forms/react-form-renderer';
-export const wizardForm = {
-    initialValues: {
-        role: 'kubernetes', // 'aws' for AWS EC2
-        verify_ssl: true
+/* Fall-back to hard-coded schemas */
+const sourceTypeSchema = t => (t.schema || temporaryHardcodedSourceSchemas[t.name]);
+
+/* return hash of form: { amazon: 'amazon', google: 'google', openshift: 'openshift' } */
+const compileStepMapper = (sourceTypes) => {
+    const names = sourceTypes.map(t => t.name);
+    return zipObject(names, names);
+};
+
+const firstStep = (sourceTypes) => ({
+    title: 'Get started with adding source',
+    name: 'step_1',
+    stepKey: 'step_1',
+    nextStep: {
+        when: 'source_type',
+        stepMapper: compileStepMapper(sourceTypes)
     },
-    schemaType: 'default',
-    showFormControls: false,
-    schema: {
-        fields: [{
-            component: componentTypes.WIZARD,
-            name: 'wizzard',
-            assignFieldProvider: true,
+    fields: [{
+        component: componentTypes.TEXT_FIELD,
+        name: 'source_name',
+        type: 'text',
+        label: 'Name'
+    }, {
+        component: componentTypes.SELECT_COMPONENT,
+        name: 'source_type',
+        label: 'Source type',
+        isRequired: true,
+        options: compileSourcesComboOptions(sourceTypes),
+        validate: [{
+            type: validatorTypes.REQUIRED
+        }]
+    }]
+});
+
+const summaryStep = () => ({
+    fields: [{
+        name: 'summary',
+        component: 'summary',
+        assignFieldProvider: true
```

Comment:
`assignFieldProvider` is no longer necessary. FieldProvider is passed to each component by default.